baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1467 @@
+		var nodeAddresses []capi.MachineAddress
+
+		c := fakeclient.NewFakeClientWithScheme(scheme)
+		machineMgr, err := newMachineManager(c, nil, nil, &tc.Machine, &tc.BMMachine, klogr.New())
+
+		if err != nil {
+			t.Error(err)
+		}
+
+		if tc.Host != nil {
+			nodeAddresses = machineMgr.nodeAddresses(tc.Host)
+			if err != nil {
+				t.Errorf("unexpected error %v", err)
+			}
+		}
+		for i, address := range tc.ExpectedNodeAddresses {
+			if address != nodeAddresses[i] {
+				t.Errorf("expected Address %v, found %v", address, nodeAddresses[i])
+			}
+		}
+	}
+}
+
+func newConfig(t *testing.T,
+	UserDataNamespace string,
+	labels map[string]string,
+	reqs []capbm.HostSelectorRequirement) (*capbm.BareMetalMachine, *corev1.ObjectReference) {
+	config := capbm.BareMetalMachine{
+		Spec: capbm.BareMetalMachineSpec{
+			Image: capbm.Image{
+				URL:      testImageURL,
+				Checksum: testImageChecksumURL,
+			},
+			UserData: &corev1.SecretReference{
+				Name:      testUserDataSecretName,
+				Namespace: UserDataNamespace,
+			},
+			HostSelector: capbm.HostSelector{
+				MatchLabels:      labels,
+				MatchExpressions: reqs,
+			},
+		},
+	}
+
+	infrastructureRef := &corev1.ObjectReference{
+		Name:       "someothermachine",
+		Namespace:  "myns",
+		Kind:       "Machine",
+		APIVersion: capi.GroupVersion.String(),
+	}
+	return &config, infrastructureRef
+}
+
+func newMachine(machineName string, bareMetalMachineName string, hostAnnotation string) *capi.Machine {
+	if machineName == "" {
+		return &capi.Machine{}
+	}
+
+	machine := &capi.Machine{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Machine",
+			APIVersion: capi.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      machineName,
+			Namespace: "myns",
+			Annotations: map[string]string{
+				HostAnnotation: hostAnnotation,
+			},
+		},
+	}
+	if bareMetalMachineName != "" {
+		machine.Spec.InfrastructureRef = v1.ObjectReference{
+			Name:       bareMetalMachineName,
+			Namespace:  "myns",
+			Kind:       "BMMachine",
+			APIVersion: capbm.GroupVersion.String(),
+		}
+	}
+	return machine
+}
+
+func newBareMetalMachine(name string,
+	ownerRef *metav1.OwnerReference,
+	spec *capbm.BareMetalMachineSpec,
+	status *capbm.BareMetalMachineStatus,
+	hostAnnotation string) *capbm.BareMetalMachine {
+
+	if name == "" {
+		return &capbm.BareMetalMachine{}
+	}
+	if spec == nil {
+		spec = &capbm.BareMetalMachineSpec{}
+	}
+	if status == nil {
+		status = &capbm.BareMetalMachineStatus{}
+	}
+	ownerRefs := []metav1.OwnerReference{}
+	if ownerRef != nil {
+		ownerRefs = []metav1.OwnerReference{*ownerRef}
+	}
+	if hostAnnotation == "undefined" {
+		return &capbm.BareMetalMachine{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "BMMachine",
+				APIVersion: capi.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            name,
+				Namespace:       "myns",
+				OwnerReferences: ownerRefs,
+			},
+			Spec:   *spec,
+			Status: *status,
+		}
+	}
```

Comment:
Instead of returning a structure everytime, you can just define some of the fields and then return he structure at the end, it removes some code duplication

Reply:
done